### ISSUE TITLE
log errors in uncaughtException handler

### ISSUE THREAD

**Issue by thadeetrompetter** [addyosmani/critical#133]
Errors are apparently being swallowed after requiring `critical`.

```javascript
throw new Error('foo'); // error is thrown to console
var critical = require('critical');
```

```javascript
var critical = require('critical');
throw new Error('bar'); // error is not thrown
```

This happens because `lib/gc.js` attaches an **unhandledException** handler to
**process**. `onException` exits with code 99, but does not log the error.

If an error is thrown somewhere in your script after having included critical
(e.g. in some other build task), you won't get to see it.

`onException` in `lib/gc.js` should log the error before calling
`process.exit`.

**Comment by thadeetrompetter** [comment]
#134

**Comment by bezoerb** [comment]
Even if the PR is already merged i guess it would be better to clear the temporary files on uncaughtException and re-throw the error.

**Comment by thadeetrompetter** [comment]
Wonder if the uncaughtException handler is needed in the first place. After all, an uncaught exception bubbling up to `process` will trigger an exit event anyway an initiate the cleanup process. The only thing you'll be missing out on would be making the process exit with code 99 from the uncaughtException handler, but does that matter?

**Comment by thadeetrompetter** [comment]
@bezoerb: Can we agree that this issue has been resolved now?

**Comment by bezoerb** [comment]
Resolved in d46c08c609fafc83a44e8328e5fd8272097d681f
